Automattic/wp-e2e-tests
# Using a single file per Gutenberg post/page tests

Following up on a something in a previous conversation....

> Right now we have 3 separate files for Gutenberg e2e tests `wp-post-editor-spec`, `wp-gutenberg-post-editor-spec` and `wp-calypso-gutenberg-post-editor-spec` that all test the same thing across different environments: Classic Editor, Gutenberg in WPCOM and dev environment /gutenberg, do you think we should see how we could use some kind of config and a single test file? I think it can work at least for the different Gutenberg setups... I think the Goal is to eventually only have tests against the actual environment `Gutenberg in Calypso` right? this kind of setup might make it easier when transitioning to that and maybe switch between testing on the different environments with minimal work in the future.

I explored a few different ways to reduce what we have but this is what I settled on so far and could use some feedback... 

 **Two main changes**

* Removed wp-calypso spec files and used environment variable `WPCALYPSO` to decide the environment and the steps to take at login
* Didn't update run script to test against this environment in the CI, I'm thinking the approach of having the ability to write code for this locally and basically switch it off until it's ready on dotcom is sufficient coverage, having tests that specifically target in the suite wpcalypso seemed a bit weird. Thoughts on this? 

To Test: 

Run `wp-gutenberg-post-editor-spec.js` and `wp-gutenberg-page-editor-spec.js` normally and it should only run against wordpress.com

Run `wp-gutenberg-post-editor-spec.js` and `wp-gutenberg-page-editor-spec.js` normally with WPCALYPSO set to true and it should only run against wpcalypso.
